apollo-router/src/plugins/telemetry/config_new/events.rs

```diff
@@ -244,49 +195,39 @@ where
     Attributes: Selectors<Request, Response, EventResponse> + Default,
     Sel: Selector<Request = Request, Response = Response> + Debug,
 {
-    pub(super) request: StandardEvent<Sel>,
-    pub(super) response: StandardEvent<Sel>,
-    pub(super) error: StandardEvent<Sel>,
+    pub(super) request: Option<StandardEvent<Sel>>,
+    pub(super) response: Option<StandardEvent<Sel>>,
+    pub(super) error: Option<StandardEvent<Sel>>,
     pub(super) custom: Vec<CustomEvent<Request, Response, EventResponse, Attributes, Sel>>,
 }
 
-impl Instrumented
-    for CustomEvents<router::Request, router::Response, (), RouterAttributes, RouterSelector>
-{
-    type Request = router::Request;
-    type Response = router::Response;
-    type EventResponse = ();
-
-    fn on_request(&self, request: &Self::Request) {
-        if self.request.level() != EventLevel::Off {
-            if let Some(condition) = self.request.condition() {
-                if condition.lock().evaluate_request(request) != Some(true) {
-                    return;
-                }
+impl CustomEvents<router::Request, router::Response, (), RouterAttributes, RouterSelector> {
```

Comment:
Also removed Instrumented here and it just required me to add `pub(crate)` to methods